c7n/filters/iamaccess.py

```diff
@@ -215,7 +218,19 @@ def check_cross_account(policy_text, allowed_accounts, everyone_only,
 
         if 'IpAddress' in s['Condition']:
             principal_ok = True
-
+            existed_srcs = s['Condition']['IpAddress']['aws:SourceIp']
+            if isinstance(existed_srcs, six.string_types):
+                existed_srcs = [existed_srcs]
+            if target_cidrs and set(target_cidrs) != set(existed_srcs):
+                for src in existed_srcs:
+                    valid_src = False
+                    for cidr in target_cidrs:
+                        if parse_cidr(cidr).__contains__(parse_cidr(src)):
```

Comment:
also the results of the parse_cidr  for src should be in a local variable out of the inner loop, else its extraneous parsing.